src/execution/operator/aggregate/physical_hash_aggregate.cpp

```diff
@@ -34,47 +34,19 @@ PhysicalHashAggregate::PhysicalHashAggregate(ClientContext &context, vector<Logi
                                              vector<GroupingSet> grouping_sets_p,
                                              vector<vector<idx_t>> grouping_functions_p, idx_t estimated_cardinality,
                                              PhysicalOperatorType type)
-    : PhysicalOperator(type, move(types), estimated_cardinality), groups(move(groups_p)),
-      grouping_sets(move(grouping_sets_p)), grouping_functions(move(grouping_functions_p)), any_distinct(false) {
+    : PhysicalOperator(type, move(types), estimated_cardinality), grouping_sets(move(grouping_sets_p)) {
 	// get a list of all aggregates to be computed
-	for (auto &expr : groups) {
-		group_types.push_back(expr->return_type);
-	}
+	idx_t groups_amount = groups_p.size();
```

Comment:
Ah yes it can indeed be `const` 👍 
As for the need to do it, I think it's still used in the "hack" for ungrouped aggregations without a `simple_update` such as `string_agg`

I'll have a look at cleaning up the hash aggregate operator when I work on the grouped variant of the distinct aggregations :)